lib/handler/redirect.c

```diff
@@ -79,10 +87,50 @@ static void redirect_internally(h2o_redirect_handler_t *self, h2o_req_t *req, h2
     h2o_send_error_503(req, "Internal Server Error", "internal server error", 0);
 }
 
+static h2o_iovec_t join_list(h2o_mem_pool_t *pool, h2o_iovec_vector_t *list, h2o_iovec_t delimiter)
+{
+    h2o_iovec_vector_t joined = (h2o_iovec_vector_t){ 0 };
+
+    assert(list->size > 0);
+    h2o_vector_reserve(pool, &joined, list->size * 2 - 1);
+
+    size_t i;
+    for (i = 0; i != list->size; ++i) {
+        if (i != 0) {
+            joined.entries[joined.size++] = delimiter;
+        }
+        joined.entries[joined.size++] = list->entries[i];
+    }
+    return h2o_concat_list(pool, joined.entries, joined.size);
+}
+
+static void build_prefix_list(h2o_iovec_t str, h2o_iovec_vector_t *dest)
+{
+    *dest = (h2o_iovec_vector_t){ 0 };
+    if (str.base == NULL) {
+        return;
+    }
+    char *start = str.base;
+    char *end = start + str.len;
+    char *p = start;
+    char *found;
+    while (p < end && (found = memchr(p, '*', end - p)) != NULL) {
```

Comment:
As I've stated below, I think that the input can be a NULL-terminated string. I think that would simplify the code, as we can use `strchr`, avoid remembering `end`.

Reply:
done in https://github.com/h2o/h2o/pull/2186/commits/7405d448a51c179ab5668139ff539e3edddfe1f8